extensions/cpp_region/PyRegion.hpp

```diff
@@ -55,6 +57,9 @@ namespace nupic
     void serialize(BundleIO& bundle);
     void deserialize(BundleIO& bundle);
 
+    virtual void write(capnp::AnyPointer::Builder& proto) const override;
```

Comment:
Some comments tying these methods to serialization/deserialization and describing the intended strategy would really help. I see serialize/deserialize as well as read/write here, and am confused about their relationship. Which ones would I use? Which ones would I need to override? Does one set of them call the other set?